ktor-utils/common/src/io/ktor/util/reflect/Type.kt

```diff
@@ -15,15 +15,35 @@ public expect val KType.platformType: Type
 
 /**
  * Ktor type information.
- * @property type: source KClass<*>
- * @property reifiedType: type with substituted generics
- * @property kotlinType: kotlin reified type with all generic type parameters.
+ * @property type Source KClass<*>
+ * @property reifiedType Type with substituted generics
+ * @property kotlinType Kotlin reified type with all generic type parameters.
  */
-public data class TypeInfo(
+public class TypeInfo(
     public val type: KClass<*>,
     public val reifiedType: Type,
     public val kotlinType: KType? = null
-)
+) {
+
+    public constructor(type: KClass<*>, kotlinType: KType) : this(
```

Comment:
I would convert this second constructor into a function. And rename `typeInfo` to `TypeInfo`. What do you think?